### ISSUE TITLE
IOperation: VB - BoundCall.DeriveArgument possibly has multiple implementation issues

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#8784]
```
                    ' Apparently the VB bound trees don't encode named arguments, which seems unnecesarily lossy.
                    Return s_argumentMappings.GetValue(argument, Function(a) If(index >= parameters.Length - 1 AndAlso parameters.Length > 0 AndAlso parameters(parameters.Length - 1).IsParamArray, New Argument(ArgumentKind.ParamArray, parameters(parameters.Length - 1), a), New Argument(ArgumentKind.Positional, parameters(index), a)))
```

It never creates an argument with kind == Named.
Also, it looks like an argument with kind ParamArray can be created for an explicitly passed array. And arguments are never created for elements of the "real" param array.

```
        /// <summary>
        /// Argument becomes an element of an array that matches a trailing C# params or VB ParamArray parameter.
        /// </summary>
        ParamArray = 0x3,

```

**Comment by AlekseyTs** [comment]
Note, it looks like an empty "real" param array is also going to be surfaced as ParamArray argument. However, if I interpret API definition correctly, "real" param array shouldn't have corresponding argument, only its elements should.

**Comment by genlu** [comment]
The first part is no longer relevant since we don't have a kind for named argument. The second part is fix, remaining issue is tracked by #18550
